### ISSUE TITLE
rewrite-target invalid for kubernetes ingress on root path

### ISSUE THREAD

**Issue by seboudry** [containous/traefik#5990]
### Do you want to request a *feature* or report a *bug*?
Bug

### What did you do?

Creating an Ingress resource on a subdomain having only one rule on `/` that point to a backend handling requests on `/app`.

```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: my-ingress
  annotations:
    kubernetes.io/ingress.class: traefik
    traefik.ingress.kubernetes.io/rewrite-target: /app
spec:
  rules:
  - host: app.my-domain.fr
    http:
      paths:
      - path: /
        backend:
          serviceName: my-app
          servicePort: http
```

### What did you expect to see?

Traefik to send all requests with URLs on domain with `app.my-domain.fr` to my backend on path:
* `/` -> `/app/`
* `/foo/` -> `/app/foo/`

Same as `nginx.ingress.kubernetes.io/rewrite-target` behavior.

### What did you see instead?

```
curl -s https://app.my-domain.fr/ | grep GET
<pre>Cannot GET /app</pre>
```
with a 404 status code.

But, request on `https://app.my-domain.fr//` (notice the additional `/` at end) give a 200 with expected response.

### traefik version

In Kubernetes, installed with Helm chart stable/traefik v1.82.3

```
traefik-helm: traefik:1.7.19
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Route rule (on dashboard):
```
PathPrefix:/;ReplacePathRegex: ^/(.*) /app$1
Host:app.my-domain.fr
```

```toml
# traefik.toml
logLevel = "info"
defaultEntryPoints = ["http","https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
  compress = true
  [entryPoints.https]
  address = ":443"
  compress = true
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      CertFile = "/ssl/tls.crt"
      KeyFile = "/ssl/tls.key"
  [entryPoints.traefik]
  address = ":8080"
[ping]
entryPoint = "http"
[kubernetes]
ingressClass = "traefik"
[traefikLog]
  format = "json"
[accessLog]
  format = "json"
[accessLog.fields]
  defaultMode = "keep"
[accessLog.fields.names]
[accessLog.fields.headers]
  defaultMode = "keep"
[accessLog.fields.headers.names]
[api]
  entryPoint = "traefik"
  dashboard = true
```

**Comment by dtomcej** [comment]
Hello @seboudry,

You gave an example of what you expected to happen:

**Comment by seboudry** [comment]
Hi @dtomcej !

Here's my working Nginx ingress definition:
```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: my-ingress-nginx
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/rewrite-target: /app/
spec:
  rules:
  - host: app.my-domain.fr
    http:
      paths:
      - backend:
          serviceName: my-app
          servicePort: http
```
So indeed the target have a trailing slash and app listen on `/app/`, will expose it later for Treafik.
And I don't have to specify `path: /` on path rules spec as it's the default value (same behaviour with or without it).


Using Traefik, if I add a trailing slash with `traefik.ingress.kubernetes.io/rewrite-target: /app/`, it doesn't seems to be used.
Indeed, `ReplacePathRegex` rule doesn't include it.
Route rule (on dashboard):
```
PathPrefix:/;ReplacePathRegex: ^/(.*) /app$1
Host:app.my-domain.fr
```

If I remove `path: /` on path rules spec, nothing is applied (regardless of trailing slash is defined on rewrite-target).
Route rule (on dashboard):
```
Host:app.my-domain.fr
```

So it seems the fix is to have the trailing slash on `rewrite-target` annotation added to `ReplacePathRegex` rule.

**Comment by dtomcej** [comment]
Hello @seboudry 

This behavior was last changed almost a year and a half ago with this PR:

**Comment by dtomcej** [comment]
There might be an edge case when the path matched is `/`

**Comment by seboudry** [comment]
Yes, It works fine when path contains something without trailing slash (`/testing` in the PR example).
But when path is on root `/` it breaks.

Just giving a hand on Traefik since this week ;)
Like it but this issue will make migration from Nginx difficult.
I'm planning to use Traefik on a couple of services and then migrate whole on v2.

**Comment by dtomcej** [comment]
Ok, then let's change this title a bit.

**Comment by dtomcej** [comment]
Ok. So it is the edge case. Will look at making a PR.

**Comment by seboudry** [comment]
Exactly!

And using the test requests from previous PR:
````
$ curl -v app.my-domain.fr/ -silent 2>&1 | grep GET

> GET / HTTP/1.1
GET /app/ HTTP/1.1

$ curl -v app.my-domain.fr/foo/ -silent 2>&1 | grep GET

> GET /foo/ HTTP/1.1
GET /app/foo/ HTTP/1.1
````

Thanks for the quick consideration of this! 👍

**Comment by dtomcej** [comment]
@seboudry, feel free to test with this PR to see if it works for you (it should).

**Comment by seboudry** [comment]
PR tested OK ;)
https://github.com/containous/traefik/pull/6005#issuecomment-565009424

**Comment by traefiker** [comment]
Closed by #6005.
